File: chor-extraction/src/main/java/ast/sp/nodes/TerminationSP.java
package ast.sp.nodes;

import ast.sp.interfaces.Behaviour;

public class TerminationSP implements Behaviour {
    private final String termination = "stop";

    @Override
    public String toString() {
        return "stop";
    }

    @Override
    public boolean findRecProcCall(String procname) {
        return false;
    }
}

File: chor-extraction/src/main/java/ast/sp/nodes/TerminationSP.kt
package ast.sp.nodes

import ast.sp.interfaces.Behaviour

class TerminationSP : Behaviour {
    private val termination = "stop"

    override fun toString(): String {
        return termination
    }

    override fun findRecProcCall(procname: String): Boolean {
        return false
    }

    override fun copy(): Behaviour {
        return TerminationSP()
    }

    override fun equals(b: Behaviour): Boolean {
        return b is TerminationSP && b.termination == termination
    }
}